src/main/java/io/github/thebusybiscuit/slimefun4/implementation/operations/CraftingOperation.java

```diff
@@ -21,6 +22,7 @@ public class CraftingOperation implements MachineOperation {
     private final ItemStack[] results;
 
     private final int totalTicks;
+    private int tickRate;
```

Comment:
Also why is this defines as a class member???
This value is not dependent on the state of instances from this class...